### PR TITLE
feat(wren-ai-service): minimize the duration for empty document and async helper for model preprocessor

### PR DESCRIPTION
**Pull request by paopa**
## Changes
1. **Document Cleaning Logic Refactoring**
   - Moved document cleaning operation to after embedding generation to minimize the duration for empty document

2. **Async Implementation Updates**
   - Added async support to `DDLChunker` methods
   - Converted `_get_ddl_commands` and `_model_preprocessor` to async methods
   - Implemented `asyncio.gather()` for parallel processing of model preprocessing

3. **Pipeline Flow Optimization**
   - Updated pipeline steps order: validate → chunk → embed → clean → write
   - Consistent implementation across all indexing pipelines (db_schema, historical_question, table_description)

4. **Logging Improvements**
   - Added custom logger setup in `dry_run_pipeline`

### CODE REVIEW

**Comment by cyyeh** [comment]
the test error is due to openai issue:

FAILED tests/pytest/pipelines/test_ask_details.py::test_generation_pipeline_producing_executable_sqls - openai.InternalServerError: <html>
[2537](https://github.com/Canner/WrenAI/actions/runs/12288904177/job/34293477579?pr=990#step:7:2538)<head><title>503 Service Temporarily Unavailable</title></head>

File: wren-ai-service/src/pipelines/common.py [hunk 1]
```diff
@@ -494,7 +494,9 @@ def dry_run_pipeline(pipeline_cls: BasicPipeline, pipeline_name: str, **kwargs):
     from src.config import settings
     from src.core.pipeline import async_validate
     from src.providers import generate_components
-    from src.utils import init_langfuse
+    from src.utils import init_langfuse, setup_custom_logger
+
+    setup_custom_logger("wren-ai-service", level_str=settings.logging_level)
```
Comment (cyyeh) [inline wren-ai-service/src/pipelines/common.py:499]: why do we need setup_custom_logger here?

**Review by cyyeh — APPROVED** [review]
LGTM, only one comment

File: wren-ai-service/src/pipelines/common.py (hunk 1 above)
Comment (paopa) [reply wren-ai-service/src/pipelines/common.py:499]: it will be more easy to see the log and adjust the level in using dry run pipeline for testing

**Pull request merged**